File: skybayfivefiftyfive_ui/src/services/RewardProgramService.js
import http from "../http-common"; 

class RewardProgramService {
  getAllRewardPrograms(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/rewardProgram/rewardPrograms`, searchDTO);
  }

  get(rewardProgramId) {
    return this.getRequest(`/rewardProgram/${rewardProgramId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/rewardProgram?field=${matchData}`, null);
  }

  addRewardProgram(data) {
    return http.post("/rewardProgram/addRewardProgram", data);
  }

  update(data) {
  	return http.post("/rewardProgram/updateRewardProgram", data);
  }
  
  uploadImage(data,rewardProgramId) {
  	return http.postForm("/rewardProgram/uploadImage/"+rewardProgramId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RewardProgramService();

File: src/main/java/com/skybayfivefiftyfive/service/RewardProgramService.java
package com.skybayfivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skybayfivefiftyfive.domain.RewardProgram;
import com.skybayfivefiftyfive.dto.RewardProgramDTO;
import com.skybayfivefiftyfive.dto.RewardProgramSearchDTO;
import com.skybayfivefiftyfive.dto.RewardProgramPageDTO;
import com.skybayfivefiftyfive.dto.RewardProgramConvertCriteriaDTO;
import com.skybayfivefiftyfive.service.GenericService;
import com.skybayfivefiftyfive.dto.common.RequestDTO;
import com.skybayfivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RewardProgramService extends GenericService<RewardProgram, Integer> {

	List<RewardProgram> findAll();

	ResultDTO addRewardProgram(RewardProgramDTO rewardProgramDTO, RequestDTO requestDTO);

	ResultDTO updateRewardProgram(RewardProgramDTO rewardProgramDTO, RequestDTO requestDTO);

    Page<RewardProgram> getAllRewardPrograms(Pageable pageable);

    Page<RewardProgram> getAllRewardPrograms(Specification<RewardProgram> spec, Pageable pageable);

	ResponseEntity<RewardProgramPageDTO> getRewardPrograms(RewardProgramSearchDTO rewardProgramSearchDTO);
	
	List<RewardProgramDTO> convertRewardProgramsToRewardProgramDTOs(List<RewardProgram> rewardPrograms, RewardProgramConvertCriteriaDTO convertCriteria);

	RewardProgramDTO getRewardProgramDTOById(Integer rewardProgramId);







}





